rusk-recovery/src/state.rs

```diff
@@ -97,13 +117,14 @@ fn generate_stake_state(
     let theme = Theme::default();
     snapshot.stakes().enumerate().for_each(|(idx, staker)| {
         info!("{} provisioner #{}", theme.action("Generating"), idx);
-        let amount = (staker.amount > 0)
-            .then(|| (staker.amount, staker.eligibility.unwrap_or_default()));
 
         let stake = StakeData {
-            amount,
+            amount: Some(StakeAmount {
+                value: staker.amount,
+                eligibility: staker.eligibility.unwrap_or_default(),
+            }),
             reward: staker.reward.unwrap_or_default(),
-            counter: 0,
+            nonce: 0,
```

Comment:
If I'm not wrong, this value is a number that gets increased right? AFAIK, nonces by definition are a random number, and they might be used along with a counter. I'd stick with counter here.

Reply:
It is indeed a number that gets strictly increased, however, I don't believe nonce just refers to a "random number". Even though it may often be used in such a way, nonce, in my understanding, refers to a number that is used only once. That's exactly the case here.

Reply:
Alright. Comment it is. I'll mention it is strictly increasing, and the best practice of simply incrementing.